doc/api/esm.md

```diff
@@ -196,664 +196,24 @@ unspecified.
-    ```
-
-   The `new` keyword isn’t required; a package’s function can return a new
-   object, or modify a passed-in object, to keep the state external to the
-   package.
-
-1. Isolate the state in one or more CommonJS files that are shared between the
-   CommonJS and ES module versions of the package. For example, if the CommonJS
-   and ES module entry points are `index.cjs` and `index.mjs`, respectively:
-
-    ```js
-    // ./node_modules/pkg/index.cjs
-    const state = require('./state.cjs');
-    module.exports.state = state;
-    ```
-
-    ```js
-    // ./node_modules/pkg/index.mjs
-    import state from './state.cjs';
-    export {
-      state
-    };
-    ```
-
-   Even if `pkg` is used via both `require` and `import` in an application (for
-   example, via `import` in application code and via `require` by a dependency)
-   each reference of `pkg` will contain the same state; and modifying that
-   state from either module system will apply to both.
-
-Any plugins that attach to the package’s singleton would need to separately
-attach to both the CommonJS and ES module singletons.
-
-This approach is appropriate for any of the following use cases:
-* The package is currently written in ES module syntax and the package author
-  wants that version to be used wherever such syntax is supported.
-* The package is stateless or its state can be isolated without too much
-  difficulty.
-* The package is unlikely to have other public packages that depend on it, or if
-  it does, the package is stateless or has state that need not be shared between
-  dependencies or with the overall application.
-
-Even with isolated state, there is still the cost of possible extra code
-execution between the CommonJS and ES module versions of a package.
-
-As with the previous approach, a variant of this approach not requiring
-conditional exports for consumers could be to add an export, e.g.
-`"./module"`, to point to an all-ES module-syntax version of the package:
-
-<!-- eslint-skip -->
-```js
-// ./node_modules/pkg/package.json
-{
-  "type": "module",
-  "main": "./index.cjs",
-  "exports": {
-    ".": "./index.cjs",
-    "./module": "./index.mjs"
-  }
-}
-```
+<!-- Anchors to make sure old links find a target -->
+<a id="esm_package_entry_points"></a>
+<a id="esm_main_entry_point_export"></a>
+<a id="esm_subpath_exports"></a>
+<a id="esm_package_exports_fallbacks"></a>
+<a id="esm_exports_sugar"></a>
+<a id="esm_conditional_exports"></a>
+<a id="esm_nested_conditions"></a>
+<a id="esm_self_referencing_a_package_using_its_name"></a>
+<a id="esm_internal_package_imports"></a>
+<a id="esm_dual_commonjs_es_module_packages"></a>
+<a id="esm_dual_package_hazard"></a>
+<a id="esm_writing_dual_packages_while_avoiding_or_minimizing_hazards"></a>
+<a id="esm_approach_1_use_an_es_module_wrapper"></a>
+<a id="esm_approach_2_isolate_state"></a>
+
+This section was moved to
+[Modules: Package configuration](package_json.html#package_json_packages).
```

Comment:
```suggestion
[Modules: The `package.json` file](package_json.html).
```

Is this URL fragment supposed to go somewhere? I can't find where to.

Reply:
That's to scroll down below the TOC

Reply:
Oh, I guess that's supposed to be a link to the heading at https://github.com/nodejs/node/pull/34748/files#diff-29fe0e955d3b8213fb3d5a0eca4773c1R11?

Would that fragment really link to that heading? I don't see how. 😕

Reply:
To be precise, it's a link to this h2 heading:

https://github.com/nodejs/node/blob/32f61aeac1ccd8cfff97d9d94cac860381120df2/doc/api/package_json.md#L11

> Would that fragment really link to that heading? I don't see how.

It does on my local branch 🤷‍♂️ I'm not sure where this behavior is defined, but don't worry `make test-doc` does check all the links to make sure none is broken.